### PR TITLE
Improve naming tests

### PR DESCRIPTION
**Pull request by BraisGabin**
This PR is a general refactor in the naming tests. Split tests in different files, simplify some, fix typos, reduce the scope of others...

### CODE REVIEW

File: detekt-rules/src/test/kotlin/io/gitlab/arturbosch/detekt/rules/naming/ConstructorParameterNamingSpec.kt [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+        }
+
+        it("should find some violations") {
+            val code = """
+                class C(val PARAM: String, private val PRIVATE_PARAM: String)
+
+                class C {
+                    constructor(PARAM: String) {}
+                    constructor(PARAM: String, PRIVATE_PARAM: String) {}
+                }
```
Comment (BraisGabin) [inline detekt-rules/src/test/kotlin/io/gitlab/arturbosch/detekt/rules/naming/ConstructorParameterNamingSpec.kt]: How does this compile? Two classes with the same name.

Comment (BraisGabin) [inline detekt-rules/src/test/kotlin/io/gitlab/arturbosch/detekt/rules/naming/ConstructorParameterNamingSpec.kt]: How does this compile? Two classes with the same name.

File: detekt-rules/src/test/kotlin/io/gitlab/arturbosch/detekt/rules/naming/NamingRulesSpec.kt [hunk 2]
```diff
@@ -1,104 +1,22 @@
...
                 data class D(val i: Int, val j: Int)
                 fun doStuff() {
                     val (_, HOLY_GRAIL) = D(5, 4)
-                    emptyMap<String, String>().forEach { _, V -> println(v) }
+                    emptyMap<String, String>().forEach { _, V -> println(V) }
                 }
             """
-            assertThat(subject.lint(code)).isEmpty()
+            assertThat(NamingRules().compileAndLint(code)).isEmpty()
         }
```
Comment (BraisGabin) [inline detekt-rules/src/test/kotlin/io/gitlab/arturbosch/detekt/rules/naming/NamingRulesSpec.kt]: I'd like to Remove NamingRulesSpec and move this to its own rule spec... But I don't know which rule is related with this test.

File: detekt-rules/src/test/kotlin/io/gitlab/arturbosch/detekt/rules/naming/ConstructorParameterNamingSpec.kt (hunk 1 above)
Comment (arturbosch) [reply detekt-rules/src/test/kotlin/io/gitlab/arturbosch/detekt/rules/naming/ConstructorParameterNamingSpec.kt]: Good question. Maybe the type checking step will first trigger redundant names?

**Pull request merged**